### PR TITLE
Bonzo Mask Durability Cooldown

### PR DESCRIPTION
**Pull request by Vixid1**
Adds durability cooldown to both fragged and non-fragged bonzo masks. Changed a bit of code for changing item's durability for less duplicate code.

image

### CODE REVIEW

**Comment by efefury** [comment]
Could you add it for Spirit Mask as well? 
This is the message: §r§6Second Wind Activated§r§a! §r§aYour Spirit Mask saved your life!§r

**Comment by Vixid1** [comment]
Added Spirit Mask support. Did some more refactoring of code, moved away from deprecated `getInternalNameForItem(ItemStack stack)` and replaced with `ItemResolutionQuery`, moved common code into functions reducing duplicate code more so than before.

image

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java [hunk 1]
```diff
@@ -153,36 +178,78 @@ private static boolean isPickaxe(String internalname) {
...
+	}
+
+	private static void setSpecificCooldown(ItemStack stack, String item) {
+		if (stack != null && stack.hasTagCompound()) {
+
+			String internalname = NotEnoughUpdates.INSTANCE.manager.createItemResolutionQuery().withItemStack(stack).resolveInternalName();
+
+			if (internalname != null) {
+				switch (item) {
+					case "Pickaxe":
```
Comment (romangraef) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java:225]: Not a big fan of using magic strings like this. Maybe use an enum, or extract the strings into constants. Other than that great PR

**Review by NopoTheGamer — APPROVED** [review]

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java (hunk 1 above)
Comment (Vixid1) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/ItemCooldowns.java]: I thought strings wasn't quite the correct thing at the time, and forgot about enums until you commented this so I've changed them to enums now.

**Review by NopoTheGamer — APPROVED** [review]

**Review by jani270 — CHANGES_REQUESTED** [review]
"Fixed Bonzo Mask and upgraded Bonzo Mask having separate cooldowns allowing for double the ability usage" 
[https://hypixel.net/threads/january-10th-some-bug-fixes.5233523/](https://hypixel.net/threads/january-10th-some-bug-fixes.5233523/)

**Review by hannibal002 — APPROVED** [review]

**Review by romangraef — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**